server/src/main/java/org/elasticsearch/action/admin/cluster/stats/AnalysisStats.java

```diff
@@ -52,6 +52,7 @@ public static AnalysisStats of(Metadata metadata) {
         final Map<String, IndexFeatureStats> usedBuiltInAnalyzers = new HashMap<>();
 
         for (IndexMetadata indexMetadata : metadata) {
+            ensureNotCancelled.run();
```

Comment:
I am not sure we provoke this line throwing in an integration test. Would be nice to provoke that to be sure it all works when integrated? Same for the MappingStats, but I suppose that provoking one of them in an IT is enough.

Reply:
These are a bit of a pain to provoke reliably - we can see that things were cancelled but it's a pain to make sure that they were cancelled before doing too much additional work since we don't really have any instrumentation to measure that work. I added unit tests in 4b08031c91bc41ee2da4c3dba139e99a18387d5e, I think these should be enough.